core-build/gulp-core-build-sass/src/SassTask.ts

```diff
@@ -76,218 +107,214 @@ export class SassTask extends GulpTask<ISassTaskConfig> {
     }
-  }
 
-  private _processFiles(
-    gulp: typeof Gulp,
-    srcPattern: string[],
-    /* tslint:disable:no-any */
-    completeCallback: (error?: string) => void,
-    postCSSPlugins: any[],
-    /* tslint:enable:no-any */
-    checkFile?: (file: gulpUtil.File) => void
-  ): NodeJS.ReadWriteStream {
-    /* tslint:disable:typedef */
-    const cleancss = require('gulp-clean-css');
-    const clipEmptyFiles = require('gulp-clip-empty-files');
-    const clone = require('gulp-clone');
-    const postcss = require('gulp-postcss');
-    const sass = require('gulp-sass');
-    const texttojs = require('gulp-texttojs');
-    const sourcemaps = require('gulp-sourcemaps');
-    /* tslint:enable:typedef */
-
-    const tasks: NodeJS.ReadWriteStream[] = [];
-
-    const srcStream: NodeJS.ReadWriteStream = gulp.src(srcPattern);
-
-    const checkedStream: NodeJS.ReadWriteStream = (checkFile ?
-      srcStream.pipe(through2.obj(
-        // tslint:disable-next-line:no-function-expression
-        function (file: gulpUtil.File, encoding: string, callback: (p?: Object) => void): void {
-          // tslint:disable-next-line:no-unused-expression
-
-          checkFile(file);
-          this.push(file);
-          callback();
+    return new Promise((resolve: (result: nodeSass.Result) => void, reject: (error: Error) => void) => {
+      nodeSass.render(
+        {
+          file: filePath,
+          importer: (url: string) => ({ file: _patchSassUrl(url) }),
+          sourceMap: this.taskConfig.dropCssFiles,
+          outFile: cssOutputPath
+        },
+        (error, result) => {
+          if (error) {
+            this.fileError(filePath, error.line, error.column, error.name, error.message);
+            reject(error);
+          } else {
+            resolve(result);
+          }
         }
-      ))
-      : srcStream);
-
-    let baseTask: NodeJS.ReadWriteStream = checkedStream;
-    if (!this.buildConfig.production) {
-      baseTask = baseTask.pipe(sourcemaps.init());
-    }
+      );
+    }).then((result: nodeSass.Result) => {
+      const options: postcss.ProcessOptions = {
+        from: filePath
+      };
+      if (result.map && !this.buildConfig.production) {
+        options.map = {
+          prev: result.map.toString()
+        };
+      }
 
-    baseTask = baseTask
-      .pipe(sass.sync({
-        importer: (url: string, prev: string, done: boolean): Object => ({ file: _patchSassUrl(url) })
-      }).on('error', function (error: Error): void {
-        sass.logError.call(this, error);
-        completeCallback('Errors found in sass file(s).');
-      }))
-      .pipe(postcss(postCSSPlugins))
-      .pipe(cleancss({
+      const plugins: postcss.AcceptedPlugin[] = [
+        ...this._postCSSPlugins,
+        ...(processAsModuleCss ? this._modulePostCssAdditionalPlugins : [])
+      ];
+      return postcss(plugins).process(result.css.toString(), options).then((postCssResult) => postCssResult);
```

Comment:
>then((postCssResult) => postCssResult); [](start = 70, length = 39)

maybe add a comment explaining why we need this.  (Is it adapting a proprietary Promise to have standard Promise typings?)